### ISSUE TITLE
Remove dependency file generator from checks

### ISSUE THREAD

**Issue by vyasr** [rapidsai/shared-workflows#46]
Currently the [checks shared workflow](https://github.com/rapidsai/shared-action-workflows/blob/branch-23.04/.github/workflows/checks.yaml) includes a job that runs dfg. As of version 1.2, dfg supports being run as a pre-commit hook, which is superior to this workflow because it can be run identically both locally and in CI. Moreover, all RAPIDS repos were migrated to using pre-commit for style checks during the GHA migration, so there's very little additional work required to add the new dfg hook. Once all repositories have been migrated to use the hook, the dfg support can be removed from the check here.

**Comment by vyasr** [comment]
CC @ajschmidt8

**Comment by ajschmidt8** [comment]
@vyasr, do we have plans to adapt this hook to support the commands below, which remove any existing generated files before re-running the `dfg` tool?

- https://github.com/rapidsai/gha-tools/blob/37f3946294647eadf5b60d8229ac396a4d913b59/tools/rapids-dependency-file-checker#L9-L15

These lines ensure that files that are no longer generated by the tool as cleaned up from the repository.

**Comment by vyasr** [comment]
Yeah I realized that was a missing bit. We could implement that as part of the Python package itself; I think it would be reasonable to do and better than relying on the action for this. The main question would be where we start searching for these files from. Since the location of dependencies.yaml is configurable, I assume that we don't want to search from the directory containing dependencies.yaml. The action assumes that it is running from the root of the repository. Do we want to bake the assumption that we are in a git repository into dfg? If so, we could use GitPython.

**Comment by ajschmidt8** [comment]
>The action assumes that it is running from the root of the repository. Do we want to bake the assumption that we are in a git repository into dfg? If so, we could use GitPython.

I'd really like to avoid adding additional dependencies to `dfg` if possible.

How does pre-commit handle this?

Can pre-commit be configured to run it from the root repository without needing `GitPython` installed?

Or will it execute from the current working directory (which may not be the root of the repository)?

I think starting from the location of `dependencies.yaml` is probably a fine idea since the generated files are all written to paths relative to that source file (and presumably, though not guaranteed, not written to directories higher up in the file tree. e.g. `../../some/path`).

**Comment by vyasr** [comment]
That's a good point, pre-commit will always run from the root of the repository so we should be able to just use that directly. I'll prototype that solution.

**Comment by vyasr** [comment]
https://github.com/rapidsai/dependency-file-generator/pull/32 adds the necessary feature to dfg. Once that is in, we should be able to specify the `--clean` option to the pre-commit hooks in each repo and move forward with the removal proposed here.

**Comment by bdice** [comment]
Repos that still have this check enabled:
- [ ] cusignal
- [ ] dask-cuda
- [ ] cucim
- [ ] kvikio
- [ ] cuxfilter
- [ ] ucx-py
- [ ] cumlprims-mg
- [ ] rapids-cmake
- [ ] cugraph-ops

**Comment by bdice** [comment]
I cleaned this up a bit in https://github.com/rapidsai/shared-action-workflows/pull/102 but we need to address the repos listed above.

**Comment by jakirkham** [comment]
Would it make sense to move this issue to [build-planning]( https://github.com/rapidsai/build-planning )?

**Comment by jameslamb** [comment]
Looked at all the non-archived repos in the `rapidsai` org with `pre-commit` configs today ([GitHub search](https://github.com/search?q=org%3Arapidsai+path%3A.pre-commit-config.yaml+NOT+is%3Aarchived&type=code&p=2)).

Of those, the following had a `dependencies.yaml` at the root of the repo but no pre-commit check running `rapids-dependency-file-generator`.

* [ ] `cucim`
  - https://github.com/rapidsai/cucim/pull/682
* [ ] `cugraph-ops` ([repo](https://github.com/rapidsai/cugraph-ops))
* [ ] `gpu-xb-ai` ([repo](https://github.com/rapidsai/gpu-xb-ai/))
* [ ] `legate-dataframe` ([repo](https://github.com/rapidsai/legate-dataframe))
* [ ] `raft-legate` ([repo](https://github.com/rapidsai/raft-legate))
* [ ] `rapids-cmake` ([repo](https://github.com/rapidsai/rapids-cmake)]

**Comment by jameslamb** [comment]
Looked at all the non-archived repos in the `rapidsai` org with `pre-commit` configs today ([GitHub search](https://github.com/search?q=org%3Arapidsai+path%3A.pre-commit-config.yaml+NOT+is%3Aarchived&type=code&p=2)).

Of those, the following had a `dependencies.yaml` at the root of the repo but no pre-commit check running `rapids-dependency-file-generator`.

* [ ] `cucim`
  - https://github.com/rapidsai/cucim/pull/682
* [ ] `rapids-cmake` ([repo](https://github.com/rapidsai/rapids-cmake)]
* [ ] some other private repos

**Comment by vyasr** [comment]
Thanks for getting started with this. It would be nice to clean up the remainder of these when you have some bandwidth!

**Comment by vyasr** [comment]
Based on the comments above, I think we can disable the check now in shared-workflows but leave the option as a no-op while we go through every repo removing the option, then once we're confident everything has it removed we can remove the check entirely from here. Am I missing anything?

**Comment by jameslamb** [comment]
I agree with that @vyasr

**Comment by vyasr** [comment]
Removing in #204

**Comment by jameslamb** [comment]
It looks to me like as of #204, this can safely be closed.
